### DISCUSSION TITLE
Inverse fuzzy/strict search

### DISCUSSION THREAD

**Discussion by gagara** [neoclide/coc.nvim#3484] [Q&A] [answered]
Is it possible to inverse search in fuzzy or strict mode in CocList?
Example:
I have a list of diagnostics. I'm not interested in any messages related to tests.
Can I somehow remove from list everything which has "test"?

**Comment by oblitum** [comment]
See `:h Lfilter`, which comes bundled to vim/nvim for filtering quickfix/location list. coc.nvim provides diagnostic on those native lists in addition to CocList, use `:CocDiagnostics` (you need to `:packadd cfilter` for `:Lfilter` to work, and you can use `:Lfilter! <stuff>` for negative search).

**Comment by gagara** [comment]
Thanks for hint. That is an option.
But by default coc-nvim does not use quickfix/locationlist for CocList and this is reasonable as for me.
I consider quickfix as "permanent" storage of locations. i.e. when I do some search and have items in CocList which I wish to save for future (aka "pin search") I just select them and send to quickfix for later. But before this it would be nice to filter them in coc-list with fuzzy search.
Would be nice to have something like:
```
FUZZY> !test
```
but maybe there is such option, however I don't find anything in doc

**Comment by oblitum** [comment]
I've never seen negative fuzzy anywhere, doesn't sound sensible.

**Comment by oblitum** [comment]
@gagara you may reserve quickfix list for that usage of yours, and use location list as usual.

**Comment by chemzqm** [comment, marked as answer]
No, my.suggestion is make your language server ignore test files.
